src/aws-metadata.js

```diff
@@ -0,0 +1,81 @@
+import request from 'request';
+import async from 'async';
+import {Logger} from './Logger';
+
+/*
+  Utility class for pulling AWS metadata that Eureka requires when registering as an Amazon instance (datacenter).
+*/
+export class AwsMetadata {
```

Comment:
Yeah it's really tricky. I generally think that acronyms should be caps (especially when it's always stylized that way), but I'm fine if we think otherwise.